### PR TITLE
Fix broken composites on archive.eclipse.org

### LINKED ISSUE #235: Archived CDT releases: composite{Artifacts,Content}.xml refer to non-existing folders
**Describe the bug**

For archived releases, the two file compositeArtifacts.xml and compositeContent.xml refer to CDT releases which are not available on https://archive.eclipse.org. I.e., the release numbers are valid, but their corresponding folders do not exist.

**To Reproduce**
1. Go to 'https://archive.eclipse.org/tools/cdt/releases/9.3/'.
2. View the raw contents of 'compositeContent.xml'
3. You will see it refers to 9.3.3, 9.3.1 and 9.3.0, but only 9.3.3 is available.

On the other hand, if you look at https://download.eclipse.org/tools/cdt/releases/10.6/ and its associated xml files, all mentioned releases are available.

**Additional context**
Do the links for releases <= 10.5 in Downloads.md need to be updated to point to https://archive.eclipse.org, or are they OK to continue to point to https://download.eclipse.org?

(I'm hoping that fixing the broken links will help me build https://github.com/wangzw/CppStyle properly, which complains about cdt9.3.1 missing.)

### PR DESCRIPTION
**Pull request by jonahgraham**
Fixes #235

### CODE REVIEW

File: releng/download/releases/9.4/compositeArtifacts.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<?compositeArtifactRepository version='1.0.0'?>
+<repository name='CDT 9.4 Update Site'
+    type='org.eclipse.equinox.internal.p2.artifact.repository.CompositeArtifactRepository' version='1.0.0'>
+  <properties size='1'>
+    <property name='p2.timestamp' value='1673888544650'/>
+  </properties>
+  <children size='5'>
```
Comment (sylt) [inline releng/download/releases/9.4/compositeArtifacts.xml:8]: Shouldn't size be 2 ?

Comment (sylt) [reply releng/download/releases/9.4/compositeArtifacts.xml:8]: As in, not 5.

Comment (jonahgraham) [reply releng/download/releases/9.4/compositeArtifacts.xml:8]: Yes. Thanks for the thorough review.

Comment (jonahgraham) [reply releng/download/releases/9.4/compositeArtifacts.xml:8]: See #248

Comment (sylt) [reply releng/download/releases/9.4/compositeArtifacts.xml:8]: Great, thank you very much for fixing this! :)

**Pull request merged**